wpiutil/src/main/native/include/wpi/Endian.h

```diff
@@ -68,9 +55,7 @@ inline value_type byte_swap(value_type value, endianness endian) {
 /// Swap the bytes of value to match the given endianness.
 template<typename value_type, endianness endian>
 inline value_type byte_swap(value_type value) {
-  if constexpr ((endian != native) && (endian != system_endianness()))
-    sys::swapByteOrder(value);
-  return value;
+  return byte_swap(value, endian);
```

Comment:
These changes revert #2454.  I think we still want to keep them?

Reply:
I patched this change (also constexpr the function that this was calling)

I left the read/write as it was. I don't see any constexpr value there, and I would imagine the optimizer would remove any overhead on calling the other inlined function, moving a template arg to a parameter.

Also decided to leave in `using make_unsigned_t = std::make_unsigned_t<value_type>;`. Keeps the patches simpler, with no cost.